### ISSUE TITLE
IPv6 - Could not bind to socket

### ISSUE THREAD

**Issue by tispokes** [xkonni/raspberry-remote#33]
Hi,

everytime i try to access the webinterface via a ipv6 device it get "could not bind to socket"-error (from the second part in the index.php (the error-message is implemented twice).
Is it possible to accept IPv6 and IPv4?

Otherwise i will try to rewrite the code of the webinterface to accept both.

Regards

**Comment by tillepille** [comment]
Did you use an IPv6 Adress in [here](https://github.com/xkonni/raspberry-remote/blob/879d5f2f01a1a435a8a5139f599b361420a5e40d/webinterface/config.php#L14) in your config?

**Comment by tispokes** [comment]
Yes and i also want to support both, v4 and v6. I could simply deactivate ipv6 on my raspi but for the future it's not the right way 🤔

**Comment by tillepille** [comment]
#15 is probably related

**Comment by tillepille** [comment]
Okay, I guess first Problem is, that the code uses `AF_INET`.  
You could use `AF_INET6` there to use it with IPv6.

Change that [here](https://github.com/xkonni/raspberry-remote/blob/879d5f2f01a1a435a8a5139f599b361420a5e40d/webinterface/index.php#L39), [here](https://github.com/xkonni/raspberry-remote/blob/879d5f2f01a1a435a8a5139f599b361420a5e40d/webinterface/index.php#L100) and around [there](https://github.com/xkonni/raspberry-remote/blob/879d5f2f01a1a435a8a5139f599b361420a5e40d/daemon.cpp#L80)...

_Long time gone since I used this code, its so messy..._  😅

The third option is a unix socket, this solves probably the problem with both protocols at once. 
Read more [here](https://www.php.net/manual/en/function.socket-create.php#refsect1-function.socket-create-parameters)

Unfortunately I don't have a Rpi laying around to test this, besides I build my own version wich works fine with both protocols..
